lib/src/screens/register_screen.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/cupertino.dart';
+
+class RegisterScreen extends StatelessWidget {
+  const RegisterScreen({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      backgroundColor: const Color(0xFFF3F4F6),
+      navigationBar: CupertinoNavigationBar(
+        backgroundColor: CupertinoColors.white,
+        border: const Border(),
+        middle: Text('Регистрация'),
+      ),
+      child: SafeArea(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: [
+                const SizedBox(
+                  height: 32,
+                ),
+                CupertinoTextField(
+                  placeholder: 'Логин',
+                  decoration: BoxDecoration(color: CupertinoColors.white),
+                  padding: const EdgeInsetsDirectional.symmetric(
+                      vertical: 20, horizontal: 16),
+                ),
+                Container(
+                  height: 1,
+                  color: Color(0xFFe0E6ED),
+                  margin: EdgeInsets.symmetric(horizontal: 16),
+                ),
+                CupertinoTextField(
+                  placeholder: 'Телефон',
+                  padding: const EdgeInsetsDirectional.symmetric(
+                      vertical: 20, horizontal: 16),
+                ),
+                Container(
+                  height: 1,
+                  color: Color(0xFFe0E6ED),
+                  margin: EdgeInsets.symmetric(horizontal: 16),
+                ),
+                CupertinoTextField(
+                  placeholder: 'Почта',
+                  padding: const EdgeInsetsDirectional.symmetric(
+                      vertical: 20, horizontal: 16),
+                ),
+                Container(
+                  height: 1,
+                  color: Color(0xFFe0E6ED),
+                  margin: EdgeInsets.symmetric(horizontal: 16),
+                ),
+                CupertinoTextField(
+                  obscureText: true,
+                  placeholder: 'Пароль',
+                  padding: const EdgeInsetsDirectional.symmetric(
+                      vertical: 20, horizontal: 16),
+                ),
```

Comment:
Добавь параметр `suffix:`, в который передадим виджет `CupertinoButton` и в него `иконку глаза`.

Функция onPressed - будет отвечать за изменение значения переменной `obscureText` на `true` или `false`.

```suggestion
                  padding: const EdgeInsetsDirectional.symmetric(
                      vertical: 20, horizontal: 16),
                  suffix: CupertinoButton(
                    child: const Icon(
                      CupertinoIcons.eye_slash,
                      color: CupertinoColors.black,
                    ),
                    onPressed: () => setState(() => obscureText = !obscureText),
                  ),
                ),
```